### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by sciabarra**
ScalaTest integration tests.

### CODE REVIEW

File: tests/src/test/scala/runtime/actionContainers/GoResourceHelpers.scala [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+      // A temporary directory for the destination files.
+      // DO NOT CREATE IT IN ADVANCE or you will get a permission denied
+      val outDir = tmpDirectoryFile("out")
+      val out = outDir.toPath.toAbsolutePath
+
+      // command to compile
+      val exe = new File(outDir, main)
+
+      val cmd = s"${dockerBin} run " +
```
Comment (rabbah) [inline tests/src/test/scala/runtime/actionContainers/GoResourceHelpers.scala]: can't you just cross compile and avoid the docker trampoline?

**Comment by rabbah** [comment]
fwiw and future reference, i would have separated this pr into two: the first to add the scala tests, and a second to remove the cram tests. it makes it easier to focus the review.

**Comment by sciabarracom** [comment]
it is just a deleted directory... I migrated those tests to Scala, 3 files. And all the test/ subdirectory gone. SImple enough (I hope).

File: tests/src/test/scala/runtime/actionContainers/GoResourceHelpers.scala (hunk 1 above)
Comment (sciabarracom) [reply tests/src/test/scala/runtime/actionContainers/GoResourceHelpers.scala]: 1. I know I have to implement all those tests. But I fist wanted to commit the infrastructure and the basic tests, then work on those.
Regarding avoiding the "docker trampoline", it is not a bug! It is a feature! 

The runtime can be used as a compiler to reduce the "cold start" problem. Tests were testing this feature you can either send the source and let the runtime compiler or use the runtime as a compiler for your sources and send the binary...

**Comment by sciabarracom** [comment]
I have removed the "test" removal from this PR. Also I can take the job of adding the other tests, just let me know. If you merge this PR I will provide in a separate one.

**Comment by rabbah** [comment]
Great thanks.

**Pull request merged**